### PR TITLE
RIA-3390: Request Home Office data event flow

### PR DESCRIPTION
**Pull request by divisathyan**
RIA-3390: Request Home Office data event flow

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-3390

### Change description ###
For handling scenario where we do not receive HO data in the validation tab, new event is created, which will allow TCW to get details from Home office

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ x] No
```

### CODE REVIEW

**Review by jkumar-moj — APPROVED** [review]
Looks good. 👍

**Review by jkumar-moj — APPROVED** [review]
👍

File: src/functionalTest/resources/scenarios/RIA-3390-request-home-office-data-success.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      "replacements": {
+        "homeOfficeReferenceNumber": "1212-0099-0062-8083",
+        "isHomeOfficeIntegrationEnabled": "Yes",
+        "contactPreferenceDescription": "Text message",
+        "appealTypeDescription": "Refusal of protection claim",
+        "homeOfficeSearchStatus": "SUCCESS"
+      }
+    }
+  }
+}
```
Comment (Srihari-Nakkala) [inline src/functionalTest/resources/scenarios/RIA-3390-request-home-office-data-success.json:32]: @divisathyan we need to cover this scenario as well **error message when event has already retrieved the data successfully***

**Review by jkumar-moj — DISMISSED** [review]
👍

File: src/functionalTest/resources/scenarios/RIA-3390-request-home-office-data-success.json (hunk 1 above)
Comment (divisathyan) [reply src/functionalTest/resources/scenarios/RIA-3390-request-home-office-data-success.json:32]: Added functional test

**Review by Srihari-Nakkala — DISMISSED** [review]
LGTM

**Review by makowalski — APPROVED** [review]
LGTM

**Pull request merged**